x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/fetch_index_patterns.test.tsx

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { renderHook, act } from '@testing-library/react-hooks';
+
+import { defaultIndexPattern } from '../../../../default_index_pattern';
+import { useApolloClient } from '../../../utils/apollo_context';
+import { mocksSource } from '../../source/mock';
+
+import { useFetchIndexPatterns, Return } from './fetch_index_patterns';
+
+const mockUseApolloClient = useApolloClient as jest.Mock;
+jest.mock('../../../utils/apollo_context');
+
+describe('useFetchIndexPatterns', () => {
+  beforeEach(() => {
+    mockUseApolloClient.mockClear();
+  });
+  test('happy path', async () => {
+    await act(async () => {
+      mockUseApolloClient.mockImplementation(() => ({
+        query: () => Promise.resolve(mocksSource[0].result),
+      }));
+      const { result, waitForNextUpdate } = renderHook<unknown, Return>(() =>
+        useFetchIndexPatterns(defaultIndexPattern)
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
```

Comment:
Is the double `await` here to ensure `useFetchIndexPatterns()` main useEffect is run, and also that its internal `fetchIndexPatterns()` returns as well?

Reply:
yes the waitForNextUpdate is working for every state change, so here I have two until getting what I wanted